src/textual/widgets/_input.py

```diff
@@ -168,11 +176,18 @@ class Input(Widget, can_focus=True):
     width = reactive(1)
     _cursor_visible = reactive(True)
     password = reactive(False)
-    max_size: reactive[int | None] = reactive(None)
     suggester: Suggester | None
     """The suggester used to provide completions as the user types."""
     _suggestion = reactive("")
     """A completion suggestion for the current value in the input."""
+    restrict = var["str | None"](None)
+    """A regular expression that must match incoming characters."""
+    type = var("text")
+    """The type of the input."""
+    max_length = var["int | None"](None)
+    """The maximum length of the input, in characters."""
```

Comment:
It's characters. If it were cells, I'd expect to see `cell_len(value)` instead of `len(value)` on the line that says:

`if self.max_length and len(value) > self.max_length:`